tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
+			    // Fill control histograms and Q vs Pos without multihits
+			    
+				if(fParameter == 6){
+					Double_t parz[3];
+                parz[0] = par->GetPar1za();
+                parz[1] = par->GetPar1zb();
+                parz[2] = par->GetPar1zc();
+
+				qbx = parz[0] + parz[1]*qbx + parz[2]*qbx*qbx;
+				}
+				
+				if (multihits[iPlane - 1][iBar - 1] < 2 /*&& qb>190. */ || 1)
+			    {
+				    fhQvsPos[iPlane - 1][iBar - 1]->Fill(posToT, qb);
+					fhQXvsPos[iPlane - 1][iBar - 1]->Fill(posToT, qbx);
+			    }
+	
+			}
+	/*	    if(fParameter==5)
+		    {
+			    auto par = fHitPar->GetModuleParAt(iPlane, iBar);
+
+			    Double_t parz[3];
+			    parz[0] = par->GetPar1za();
+			    parz[1] = par->GetPar1zb();
+			    parz[2] = par->GetPar1zc();
+			    //parq[3] = par->GetPold();
+
+
+
+			    Double_t qb = 0.;
+			    if (fTofdSmiley)
+			    {
+				    qb = TMath::Sqrt(top_tot * bot_tot);///(parq[0] + parq[1] * posToT + parq[2] * pow(posToT, 2) + parq[3] * pow(posToT, 3));
+				    qb = qb *
+					    fTofdQ; // theory says: dE ~ Z^2 but we see quenching -> just use linear and fit the rest
+				    // std::cout<< " hola30 " << qb << " , " << fTofdQ << std::endl;
+			    }
+
+			    // Fill control histograms and Q vs Pos without multihits
+			    if (multihits[iPlane - 1][iBar - 1] < 2 && (qb > 5.6 && qb < 6.7.))
+			    {
+				    //    std::cout<< " h: " << posToT << " , " << qb << std::endl;
+				    fhZoffset[iPlane - 1][iBar - 1]->Fill(parz[0]+parz[1]*qb);
+			    }
+
+
+
+		    }
+*/
+		    ++top_i;
+		    ++bot_i;
+
+		    // Increment events
+		    fNEvents += 1;
+		}
+		else if (dt < 0 && dt > -c_range_ns / 2)
+		{
+			++top_i;
+		}
+		else
+		{
+			++bot_i;
+		}
+	    }
+	}
+    }
+	void R3BTofDCal2HitPar::SetHistogramsNULL(Int_t iPlane, Int_t iBar){
+		fhTdiff[iPlane - 1] = NULL;
+		fhTsync[iPlane - 1] = NULL;
+		fhTofsync[iPlane - 1] = NULL;
+		fh_tofd_TotPm[iPlane - 1][0] = NULL;
+		fh_tofd_TotPm[iPlane - 1][1] = NULL;
+		fh_tofd_TotPm[iPlane - 1][2] = NULL;
+		fh_tofd_TotPm1[iPlane - 1] = NULL;
+		fh_tofd_TotPm2[iPlane - 1] = NULL;
+		fhLogTot1vsLogTot2[iPlane - 1][iBar - 1] = NULL;
+		fhSqrtQvsPosToT[iPlane - 1][iBar - 1] = NULL;
+		fhQvsPos[iPlane - 1][iBar - 1] = NULL;
+		fhQXvsPos[iPlane - 1][iBar - 1] = NULL;
+		fhTot1vsPos[iPlane - 1][iBar - 1] = NULL;
+		fhTot2vsPos[iPlane - 1][iBar - 1] = NULL;
+		fh1_tofsync[iPlane - 1][iBar - 1] = NULL;
+		fh1_walk_t[iPlane - 1][iBar - 1] = NULL;
+		fh1_walk_b[iPlane - 1][iBar - 1] = NULL;
+
+	}
+    void R3BTofDCal2HitPar::CreateHistograms(Int_t iPlane, Int_t iBar)
+    {
+	    Double_t max_charge = fMaxQ;
+
+	    if (NULL == fhTdiff[iPlane - 1])
+	    {
+		    char *strName1;
+			strName1 = (char*)malloc(255);
+		    char *strName2;
+			strName2 = (char*)malloc(255);
+		    sprintf(strName1, "Time_Diff_Plane_%d", iPlane);
+		    sprintf(strName2, "Time Diff Plane %d", iPlane);
+		    fhTdiff[iPlane - 1] = new TH2F(strName1, strName2, 50, 0, 50, 4000, -20., 20.);
```

Comment:
Could use a string instead. Would save 4 lines of code per Histogram declaration